### DISCUSSION TITLE
Multiplatform Support

### DISCUSSION THREAD

**Discussion by petehellyer** [AppIntro/AppIntro#1212] [Ideas]
Hi there,

I'm very interested in using AppIntro, and with Compose Multiplatform now in Beta for iOS, I'm also heavily pushing towards the adoption of Compose Multiplatform for our apps.

So, the question is, how suitable is the current code-base for KMP and Compse Multiplatform? I'm willing to put some time into helping to port as well, but any pointers in the right direction as to what would be core to change would be welcome!

Cheers!

**Comment by cortinico** [comment]
Hey @petehellyer 
Thanks for reaching out to us!

> So, the question is, how suitable is the current code-base for KMP and Compse Multiplatform? I'm willing to put some time into helping to port as well, but any pointers in the right direction as to what would be core to change would be welcome!

The first step would be to migrate the library from views to compose 👍
If you're up for it, we (@paolorotolo + me) are more than happy to support.

Currently the library uses viewpager which is not KMP compatible, so moving to Compose will definitely open more doors here.

**Comment by paolorotolo** [comment]
Hi @petehellyer! This is definitely something we should do!
As @cortinico said, we should first adapt the existing code to support Compose. I've created a [Project here](https://github.com/orgs/AppIntro/projects/1) to follow the progress. I'll be adding some tickets to get this party started 😄
